client/main/options.js

```diff
@@ -2,166 +2,178 @@
 
 'use strict';
 
-const Options = function() {
+const SuperClass = require('../common/superclass');
+const _ = require('underscore');
 
-};
+const OptionType = {
 
-Options.toPB = function(options, Messages) {
-    let names = [ ];
-    let optionsPB = [ ];
+    create: function(template, value) {
+        var constructor = OptionType[template.type + "Type"];
+        if ( ! constructor)
+            constructor = OptionType.Option;
 
-    for (let name in options) {
-        names.push(name);
-        optionsPB.push(_toPB(options[name], Messages));
+        return new constructor(template, value);
     }
-
-    let child = new Messages.AnalysisOptions();
-    child.setOptions(optionsPB);
-    child.setNames(names);
-    child.setHasNames(true);
-
-    return child;
 };
 
-const _toPB = function(value, Messages) {
-
-    if (value === true) {
-        let option = new Messages.AnalysisOption();
-        option.setO(Messages.AnalysisOption.Other.TRUE);
-        option.type = 'o';
-        return option;
-    }
-    else if (value === false) {
-        let option = new Messages.AnalysisOption();
-        option.setO(Messages.AnalysisOption.Other.FALSE);
-        option.type = 'o';
-        return option;
-    }
-    else if (value === null) {
-        let option = new Messages.AnalysisOption();
-        option.setO(Messages.AnalysisOption.Other.NULL);
-        option.type = 'o';
-        return option;
-    }
-    else if (Number.isInteger(value)) {
-        let option = new Messages.AnalysisOption();
-        option.setI(value);
-        option.type = 'i';
-        return option;
-    }
-    else if (typeof value === 'number') {
-        let option = new Messages.AnalysisOption();
-        option.setD(value);
-        option.type = 'd';
-        return option;
-    }
-    else if (typeof value === 'string') {
-        let option = new Messages.AnalysisOption();
-        option.setS(value);
-        option.type = 's';
-        return option;
-    }
-    else if (Array.isArray(value)) {
-
-        let options = new Array(value.length);
+OptionType.Option = function(template, value) {
+    this._template = template;
 
-        let arrayify = false;
-        for (let i = 0; i < value.length; i++) {
-            if (Array.isArray(value[i])) {
-                arrayify = true;
-                break;
-            }
-        }
+    this.update = function(value) {
```

Comment:
would `setValue()` be a bit clearer?
`update()` suggests to me slightly different behaviour.